### ISSUE TITLE
Refactor Reporter to use generic batching capabilities

### ISSUE THREAD

**Issue by trevrosen** [appoptics/appoptics-api-go#17]
(blocked by #4)

- `Reporter` has its own batching implementation. Refactor to use `MeasurementsBatch` with `BatchPersister`
- Extend the batching capabilities to handle `Reporter`'s need for batches that have a calculated time based on `Reporter.outputMeasurementIntervalSeconds`

**Comment by trevrosen** [comment]
Closing in favor of using the Go metrics lib
